lib/puppet/provider/iis_application_pool/webadministration.rb

```diff
@@ -12,10 +12,7 @@
   mk_resource_methods
 
   def exists?
-    inst_cmd = "Get-WebConfiguration -Filter '/system.applicationHost/applicationPools/add' | ? { $_.Name -eq '#{@resource[:name]}' }"
```

Comment:
Is the time cost here from opening a runspace, or from the query? If from the query we can further filter:

`Get-WebConfiguration -Filter "/system.applicationHost/applicationPools/add[@name='foo']//."`